dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -294,6 +295,14 @@ def login
   end
 
   private def register_new_user(user, provider)
+    # Disallow sign up with email addresses from disallowed domains
+    if user.email.present?
+      domain = user.email.include?('@') ? user.email.split('@', 2)[1] : nil
```

Comment:
This seems too deep/late for these kind of validation checks, and when I look at where `register_new_user` is called, I can see earlier code that expect `user.email` to exist. Can this be validated in one place, earlier in the request?

Reply:
I ask, because if we know that `user.email` exists and is an email address, then you can just do 

```
domain = user.email.split('@')[-1]
```

Reply:
Yeah that's valid, let me verify we are doing an email validation earlier. I know we are if this is coming from the email password signup form, and I imagine we should be for OAuth reponses.